### PR TITLE
Feat/tavily

### PR DESCRIPTION
**Pull request by pulvedu**
## Description

This PR introduces Tavily integration to Strands Tools - a comprehensive web search, extraction, crawling, and mapping toolkit specifically optimized for AI agents. Tavily provides real-time web search capabilities with intelligent content filtering and extraction.

### New Tools Added
This PR adds 4 new Tavily tools with both synchronous and asynchronous implementations:

- tavily_search / tavily_search_async - Real-time web search optimized for AI agents with advanced content filtering, news search, domain management, and AI-generated answers

- tavily_extract / tavily_extract_async - Extract clean, structured content from web pages with noise removal and multiple output formats

- tavily_crawl / tavily_crawl_async - Intelligent website crawling starting from a base URL with filtering, extraction, and parallel exploration

- tavily_map / tavily_map_async - Website structure mapping and URL discovery without content extraction for faster site analysis

## Type of Change

New Tool

## Testing

- Created comprehensive test examples in tavily-test.py demonstrating all 4 tools
- Tested both synchronous and asynchronous implementations
- Verified rich console formatting and error handling
- Validated API integration with proper authentication
- Confirmed all parameter validation and edge cases

[x] I ran `hatch run prepare`

## Checklist
- [x] I have read the CONTRIBUTING document
- [x] I have added any necessary tests that prove my fix is effective or my feature works
- [x] I have updated the documentation accordingly
- [x] I have added an appropriate example to the documentation to outline the feature
- [x] My changes generate no new warnings
- [x] Any dependent changes have been merged and published

----

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/strands_tools/tavily.py [hunk 1]
```diff
@@ -0,0 +1,1125 @@
...
+        return {"status": "error", "content": [{"text": "Connection error. Please check your internet connection."}]}
+    except ValueError as e:
+        return {"status": "error", "content": [{"text": str(e)}]}
+    except Exception as e:
+        logger.error(f"Unexpected error in tavily_crawl: {str(e)}")
+        return {"status": "error", "content": [{"text": f"Unexpected error: {str(e)}"}]}
+
+
+@tool
+def tavily_map(
```
Comment (jer96) [inline src/strands_tools/tavily.py]: we recommend all tools be defined as async. async tools can be invoked by an agent in a synchronous context, this is all abstracted away behind the `Agent` interface. 

here is an example PR of how this change was applied to the `A2A` client tool: https://github.com/strands-agents/tools/pull/165

**Comment by pulvedu** [comment]
@jer96, I made the changes to make the tool functions async only, please let me know if anything else needs to be changed
